npm/pkg/dataplane/ipsets/ipsetmanager_linux.go

```diff
@@ -1,21 +1,193 @@
 package ipsets
 
 import (
+	"bytes"
 	"fmt"
+	"os"
+	"strings"
 
-	"github.com/Azure/azure-container-networking/npm/util/errors"
-	"k8s.io/klog"
+	"github.com/Azure/azure-container-networking/npm/util"
+	kexec "k8s.io/utils/exec"
 )
 
+// TODO look at current errors in kusto
+
+// TODO eventually, have multiple retries and start at spot based on line number in error?
+const maxRetryCount = 1
+
+type ipsetRestoreFile struct {
```

Comment:
can we abastract this part out since this will be same for both ipsets and iptables